Northstar.Coop/mod/scripts/vscripts/sp/sp_sewer_common.nut

```diff
@@ -0,0 +1,285 @@
+
+	TimerInit( "Nag_BTTakingToxicDamage", 10.0 )
+	TimerInit( "Nag_PlayerTitanTakingToxicDamage", 5.0 )
+	TimerInit( "Nag_PlayerPilotInToxicWaste", 5.0 )
+	TimerInit( "ToxicWasteDamage", TOXIC_WASTE_DMG_DEBOUNCE_TIME )
+}
+
+
+void function SewerCommon_EntitiesDidLoad()
+{
+	ToxicTriggersSetup()
+}
+
+
+void function SewerCommon_PlayerDidRespawn( entity player )
+{
+	thread ToxicWaste_GeigerCounter_AndPlayerEffects( player )
+}
+
+
+void function SewersCommon_TitanBecomesPilot( entity pilot, entity titan )
+{
+	Assert( pilot.IsPlayer() )
+	PlayerTitan_StayPut( pilot )
+}
+
+
+void function ToxicTriggersSetup()
+{
+	array<entity> hurtTrigs = GetEntArrayByClass_Expensive( "trigger_hurt" )
+
+	foreach ( trig in hurtTrigs )
+	{
+		string trigName = trig.GetScriptName()
+
+		if ( trigName.find( "trig_toxic_waste" ) != null )
+		{
+			file.toxicWasteTrigs.append( trig )
+			trig.SetNearbyRadius( TOXIC_AREA_NEARBY_DIST )
+		}
+
+		if ( trigName.find( "SewerArena_sludge_wall_hurt_trigger" ) != null )
+		{
+			file.toxicWasteTrigs.append( trig )
+			trig.SetNearbyRadius( TOXIC_AREA_NEARBY_DIST )
+		}
+	}
+}
+
+
+void function Sewer_PlayerTookDamage( entity ent, var damageInfo )
+{
+	if ( !IsAlive( ent ) )
+		return
+
+	entity attacker = DamageInfo_GetAttacker( damageInfo )
+	if ( attacker.GetClassName() != "trigger_hurt" )
+		return
+
+	// ---- Toxic Waste ----
+	entity toxicWasteDamageTrig = null
+	foreach ( trig in file.toxicWasteTrigs )
+	{
+		if ( attacker == trig )
+		{
+			toxicWasteDamageTrig = attacker
+			break
+		}
+	}
+
+	if ( toxicWasteDamageTrig != null )
+	{
+		if ( !TimerCheck( "ToxicWasteDamage" ) )
+		{
+			DamageInfo_SetDamage( damageInfo, 0 )
+			return
+		}
+
+		if ( ent.IsPlayer() && ent.IsTitan() )
+		{
+			DamageInfo_SetDamage( damageInfo, 0 )
+			return
+		}
+
+		DamageInfo_SetDamage( damageInfo, TOXIC_WASTE_DMG_MIN_PILOT )
+
+		if ( ent.IsPlayer() && TimerCheck( "Nag_PlayerPilotInToxicWaste" ) )
+		{
+			Remote_CallFunction_NonReplay( ent, "ServerCallback_SewersToxicSludgeAlarmSeq" )
+			TimerReset( "Nag_PlayerPilotInToxicWaste" )
+		}
+
+		TimerReset( "ToxicWasteDamage" )
+	}
+}
+
+
+function ToxicWaste_GeigerCounter_AndPlayerEffects( entity player )
+{
+	if ( !IsAlive( player ) )
+		return
+
+	player.Signal( "ToxicWasteGeigerCounter_Stop" )
+	player.EndSignal( "ToxicWasteGeigerCounter_Stop" )
+	player.EndSignal( "OnDestroy" )
+
+	string geigerSFX_1 = "Pilot_GeigerCounter_Warning_LV1"
+	string geigerSFX_2 = "Pilot_GeigerCounter_Warning_LV2"
+	string geigerSFX_3 = "Pilot_GeigerCounter_Warning_LV3"
+
+	int lastGeigerLevel = 0
+	string currSFX = ""
+
+	OnThreadEnd(
+	function() : ( player, geigerSFX_1, geigerSFX_2, geigerSFX_3 )
+		{
+			if ( IsValid( player ) )
+			{
+				// breaks the game
+				try{
+				player.StopSoundOnEntity( geigerSFX_1 )
+				player.StopSoundOnEntity( geigerSFX_2 )
+				player.StopSoundOnEntity( geigerSFX_3 )
+				}
+				catch( aaaaaaaaa )
+				{
+					print( aaaaaaaaa )
+				}
```

Comment:
what's the error here?

Reply:
I forgor

Reply:
it works so I wouldn't want to touch it.